### PR TITLE
feat: Validate AWS Tag Structure is not `[]`

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

Adds a test to validate that the JSON object is not `[]`

### CODE REVIEW

File: plugins/source/aws/client/testing.go [hunk 1]
```diff
@@ -150,3 +153,36 @@ func validateNoEmptyColumnsExcept(t *testing.T, tables schema.Tables, messages m
...
+	tablesWithIssues := []string{}
+	for _, table := range tables.FlattenTables() {
+		if table.Column("tags") == nil {
+			continue
+		}
+
+		records := messages.GetInserts().GetRecordsForTable(table)
+		for _, resource := range records {
+			for colIndex, arr := range resource.Columns() {
+				if table.Columns[colIndex].Name != "tags" {
```
Comment (candiduslynx) [inline plugins/source/aws/client/testing.go:168]: get index by name & use only that?
https://pkg.go.dev/github.com/cloudquery/plugin-sdk/v4@v4.38.0/schema#ColumnList.Index

File: plugins/source/aws/client/testing.go [hunk 2]
```diff
@@ -150,3 +153,36 @@ func validateNoEmptyColumnsExcept(t *testing.T, tables schema.Tables, messages m
...
+					continue
+				}
+				for i := 0; i < arr.Len(); i++ {
+					if arr.IsValid(i) {
```
Comment (candiduslynx) [inline plugins/source/aws/client/testing.go:172]: less indent could be better via
```go
if arr.IsNull(i) { continue }
```

**Review by candiduslynx — CHANGES_REQUESTED** [review]

**Review by candiduslynx — APPROVED** [review]

File: plugins/source/aws/client/testing.go [hunk 3]
```diff
@@ -150,3 +153,27 @@ func validateNoEmptyColumnsExcept(t *testing.T, tables schema.Tables, messages m
...
+	for _, table := range tables.FlattenTables() {
+		index := table.Columns.Index("tags")
+		if index == -1 {
+			continue
+		}
+		records := messages.GetInserts().GetRecordsForTable(table)
+		for _, resource := range records {
+			arr := resource.Column(index)
+			for i := 0; i < arr.Len(); i++ {
+				val := arr.GetOneForMarshal(i).(json.RawMessage)
```
Comment (candiduslynx) [inline plugins/source/aws/client/testing.go:168]: Won't it panic on nil?

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: plugins/source/aws/client/testing.go (hunk 3 above)
Comment (bbernays) [reply plugins/source/aws/client/testing.go:168]: It should never be null. If it was empty the test would have failed at this point

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**